test/test_parse_perceived_health.rb

```diff
@@ -0,0 +1,382 @@
+require 'minitest/autorun'
+load 'test/test_harness.rb'
+
+include Harness
+
+class TestPerceiveHealth < Minitest::Test
+  def setup
+    $server_buffer.clear
+    $history.clear
+  end
+
+  def teardown
+    @test.join if @test
+  end
+
+  def assert_poisoned
+    proc { |health| assert_equal(true, health['poisoned'], 'Person is poisoned but reported as not poisoned') }
+  end
+
+  def assert_not_poisoned
+    proc { |health| assert_equal(false, health['poisoned'], 'Person is not poisoned but reported as poisoned') }
+  end
+
+  def assert_diseased
+    proc { |health| assert_equal(true, health['diseased'], 'Person is diseased but reported as not diseased') }
+  end
+
+  def assert_not_diseased
+    proc { |health| assert_equal(false, health['diseased'], 'Person is not diseased but reported as diseased') }
+  end
+
+  def assert_no_parasites
+    proc { |health| assert_empty(health['parasites'], 'Person parasite free but is reported as host to parasites') }
+  end
+
+  def assert_parasites(expected_number)
+    # Parasites are always severity 1 today, so just select the length of that value array
+    proc { |health| assert_equal(expected_number, health['parasites'][1].count, 'Person has wrong number of parasites.') }
```

Comment:
Thoughts on flattening the map like in `assert_wound_count` method?
```suggestion
    proc { |health| assert_equal(expected_number, health['parasites'].values.flatten.count, 'Person has wrong number of parasites.') }
```

Reply:
I can - however since they're always severity one, anything that caused non-severity 1 parasites would be considered an unintended change right now and we might want the test to fail?

Reply:
That works, too, thanks